src/expr/src/table_function/mod.rs

```diff
@@ -0,0 +1,64 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use itertools::Itertools;
+use risingwave_common::array::{ArrayRef, DataChunk};
+use risingwave_common::types::DataType;
+use risingwave_pb::expr::TableFunction as TableFunctionProst;
+
+use super::Result;
+use crate::expr::{build_from_prost as expr_build_from_prost, BoxedExpression};
+
+mod generate_series;
+use generate_series::*;
+mod unnest;
+use unnest::*;
+
+/// Instance of a table function.
+///
+/// A table function takes a row as input and returns a table. It is also known as Set-Returning
+/// Function.
+pub trait TableFunction: std::fmt::Debug + Sync + Send {
+    fn return_type(&self) -> DataType;
```

Comment:
It seems PG returns a Row type. Interesting 👀

```sql
postgres=# CREATE TABLE foo (fooid int, foosubid int, fooname text);
INSERT INTO foo VALUES (1, 1, 'Joe'), (1, 2, 'Ed'), (2, 1, 'Mary');

CREATE FUNCTION getfoo(int) RETURNS SETOF foo AS $$
    SELECT * FROM foo WHERE fooid = $1;
$$ LANGUAGE SQL;

SELECT * FROM getfoo(1);
CREATE TABLE
INSERT 0 3
CREATE FUNCTION
 fooid | foosubid | fooname
-------+----------+---------
     1 |        1 | Joe
     1 |        2 | Ed
(2 rows)

postgres=# explain select * from getfoo(1);
                           QUERY PLAN
----------------------------------------------------------------
 Function Scan on getfoo  (cost=0.25..10.25 rows=1000 width=40)
(1 row)

postgres=# select getfoo(1);
  getfoo
-----------
 (1,1,Joe)
 (1,2,Ed)
(2 rows)

postgres=# explain select getfoo(1);
                    QUERY PLAN
--------------------------------------------------
 ProjectSet  (cost=0.00..5.27 rows=1000 width=32)
   ->  Result  (cost=0.00..0.01 rows=1 width=0)
(2 rows)
```

Reply:
Will this be fixed in this PR?

Reply:
I think we don't need to fix anything? Currently we don't have TF returning more than one cols. When we have in the future, it will be represented by a nested type. So the `TableFunction` trait doesn't need to be changed/